### ISSUE TITLE
Introduce unique-label structure

### ISSUE THREAD

**Issue by jo-mueller** [allysonryan/napari-toska#16]
I propose to format the toska skeleton structure in such a way that every object (be it a branch/chain/end-point) carries a unique label. An associated pandas dataframe would then carry the information about what kind of object each label refers to. The dataframe would look like this:

```
| Label | Object-type |
| --- | --- |
| 1 | Endpoint |
| 2 | Chain |
| 3 |Chain |
| ... | ... |
```

This would require the dataframe and the image to always stay with each other - the easiest way to encapsulate this in code is probably for a function that generates this structure to return a `napari.layer.Layer`, which comes with an associated `layer.features` table that can hold this information.

The functions we created previously (`generate_labelled_skeleton` and so on) could then simply reformat this structure into the images we currently return (i.e., just a numpy array)

Another advantage is that further properties (`skeleton_id`, `branch_id`, etc.) can also be encapsulated in this table so that it is similarly easy to create a branch image, a labeled skeleton image, and all the other ones we have been working with.

*Edit*: Maybe a cheap way to get everything together would be to introduce a `ToskaSkeleton` class that inherits from the `napari.layers.Labels` class. That way, a Toska skeleton could always retain the information stored in the labels layer and could be added to a napari viewer natively. At the same time, other properties (i.e., adjacency matrices, an nx graph object, the neighborhood kind) could be stored in the layer's metadata and ensure that they would always be passed on together.

**Comment by allysonryan** [comment]
I agree that a new class is a good idea. Would probably simplify indexing results for statistical analysis as well. I am working on a new package of functions for more accurate visualization of networks created from image data. We can try to make them compatible.
